glfs.c

```diff
@@ -329,6 +329,7 @@ static int parse_imagepath(char *cfgstring, gluster_server **hosts)
 
 fail:
 	gluster_free_server(entry);
+	*hosts = NULL;
```

Comment:
At the last of gluster_free_server(), there has one "hosts = NULL;", I think it intended to do this, but it just pasre one pionter of entry, may it should be "&entry" ?